### PR TITLE
feat(system.networkInterface): add networkInterface faker

### PR DESCRIPTION
**Pull request by iandrewt**
I built this to use in one of my own projects, based on the [SystemD predictable network interface device names schema](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/ch-consistent_network_device_naming) (although this is likely the antithesis of that 😛 ). Figured it might be worth sharing, but if it's too niche then no hard feelings 😄

### CODE REVIEW

File: src/modules/system/index.ts [hunk 1]
```diff
@@ -171,4 +179,69 @@ export class System {
...
+   */
+  networkInterface(
+    options: {
+      interfaceType?: typeof commonInterfaceTypes[number];
+      interfaceSchema?: keyof typeof commonInterfaceSchemas;
+    } = {}
+  ): string {
+    const {
+      interfaceType = this.faker.helpers.arrayElement(commonInterfaceTypes),
+      interfaceSchema = this.faker.helpers.arrayElement(
```
Comment (ST-DDT) [inline src/modules/system/index.ts:204]: You could also use 
https://fakerjs.dev/api/helpers.html#objectKey

Comment (iandrewt) [reply src/modules/system/index.ts:204]: Oh that's nifty, update incoming

**Comment by ST-DDT** [comment]
**Preview:** ✅ 

grafik

**Review by ST-DDT — DISMISSED** [review]

**Review by Shinigami92 — DISMISSED** [review]
I'm personally not really happy what prettier does for the formatting here but that is not the authors problem
Also the test regex are very heavy, but they work
The PR works, so I will approve

File: test/system.spec.ts [hunk 2]
```diff
@@ -46,6 +47,29 @@ describe('system', () => {
...
+              interfaceSchema ? ` interfaceSchema ${interfaceSchema}` : ''
+            }`,
+            {
+              interfaceType,
+              interfaceSchema,
+            }
+          );
+        }
+      }
+    });
```
Comment (Shinigami92) [inline test/system.spec.ts:72]: @ST-DDT please check if I made that right (was my first time working with that new factory)
feedback: you cannot easily click anymore on the function that is tested to get to the definition of it :confused: 
luckily I can jump to it via click on one of the option params, but for functions without options it's not possible

Comment (ST-DDT) [reply test/system.spec.ts:72]: I would probably use `with ${options}` for simplicity sake, but this works for me as well.

**Review by ST-DDT — APPROVED** [review]

File: test/system.spec.ts (hunk 2 above)
Comment (Shinigami92) [reply test/system.spec.ts:72]: Oh I thought using `${options}` would result in `'[object Object]'`

Comment (ST-DDT) [reply test/system.spec.ts:72]: Well, you could still jsonify it.

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**